lighthouse-core/config/config.js

```diff
@@ -341,17 +351,47 @@ class Config {
   }
 
   /**
-   * Filter out any unrequested categories from the categories object.
+   * Filter out any unrequested categories or audits from the categories object.
    * @param {!Object<string, {audits: !Array<{id: string}>}>} categories
-   * @param {Array<string>=} categoryIds
+   * @param {!Array<string>=} categoryIds
+   * @param {!Array<string>=} auditIds
    * @return {!Object<string, {audits: !Array<{id: string}>}>}
    */
-  static filterCategories(oldCategories, categoryIds = []) {
+  static filterCategoriesAndAudits(oldCategories, categoryIds = [], auditIds = []) {
     const categories = {};
 
+    // warn if the category is not found
+    categoryIds.forEach(category => {
+      if (!oldCategories[category]) {
+        log.warn('config', `unrecognized category: ${category}`);
```

Comment:
need a clearer way of saying this, since users will see it without context. Not sure what would be good though... maybe just ``unrecognized category in `onlyCategories`: ${category}``?